src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -447,6 +444,9 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
 
         // If we have any out of date connections, dispose of them
         disposedControllers.forEach((controller) => {
```

Comment:
@rchiodo you mentioned not needing to save registeredControllers, but we both add (a few lines above here) and remove from this set after the initial load, so I think that we do need to save it off and not just use the promise results.

Reply:
Then it seems that getNotebookControllers is incorrect. If somebody else calls that function they won't get the registered controllers as the list is modified outside of that function.

Reply:
Bleh, there is indeed a grody discrepancy there.

So maybe this:

- getNotebookControllers => loadNotebookControllers - Responsible for performing the initial load of controllers. Returns an empty promise so it can be awaited (tests use this) but just assigns results to the registered list and doesn't return the list.
- getNotebookControllers - New function. Non-async to just return the current set of registered controllers. Believe that we only would need this for tests.

Reply:
Looks like the remote test I reenabled is failing anyways so gotta investigate.